src/main/java/de/numcodex/sq2cql/model/Mapping.java

```diff
@@ -38,24 +40,32 @@ public Mapping(TermCode key, String resourceType, String valueFhirPath, String v
         this.attributeMappings = (attributeMappings == null ? Map.of() : attributeMappings.stream()
                 .collect(Collectors.toMap(AttributeMapping::key, Function.identity())));
         this.timeRestrictionPath = timeRestrictionPath;
+        this.codeFhirPath = codeFhirPath;
+        this.primaryCode = primaryCode;
     }
 
     public static Mapping of(TermCode key, String resourceType) {
-        return new Mapping(key, resourceType, "value", null, List.of(), List.of(), null);
+        return new Mapping(key, resourceType, "value", null, List.of(), List.of(), null, null, null);
     }
 
     public static Mapping of(TermCode concept, String resourceType, String valueFhirPath) {
-        return new Mapping(concept, resourceType, valueFhirPath, null, List.of(), List.of(), null);
+        return new Mapping(concept, resourceType, valueFhirPath, null, List.of(), List.of(), null, null, null);
     }
 
     public static Mapping of(TermCode concept, String resourceType, String valueFhirPath, String valueType) {
-        return new Mapping(concept, resourceType, valueFhirPath, valueType, List.of(), List.of(), null);
+        return new Mapping(concept, resourceType, valueFhirPath, valueType, List.of(), List.of(), null, null, null);
     }
 
     public static Mapping of(TermCode key, String resourceType, String valueFhirPath, String valueType, List<Modifier> fixedCriteria,List<AttributeMapping> attributeMappings) {
         return new Mapping(key, resourceType, valueFhirPath == null ? "value" : valueFhirPath, valueType,
             fixedCriteria == null ? List.of() : List.copyOf(fixedCriteria),
-            attributeMappings, null);
+            attributeMappings, null, null, null);
+    }
+
+    public static Mapping of(TermCode key, String resourceType, String valueFhirPath, String valueType, List<Modifier> fixedCriteria, List<AttributeMapping> attributeMappings, String timeRestrictionPath) {
```

Comment:
At this point we should switch from `of` to builders. There are just to much arguments here.

Reply:
I assume you mean the builder pattern. I agree but this will require refactoring all test cases. I would suggest we postpone this.